### ISSUE TITLE
[Andoroid] PushNotificationInfo fromMap can't hanlde notification from ANH

### ISSUE THREAD

**Issue by hyounoo** [Azure/Communication#240]
**Describe the bug**
Parse notification message function in PushNotificationInfo.class from com.azure.android:azure-communication-calling:1.0.0-beta.9 has Firebase-Messaging-Intent-Handle exception.

**To Reproduce**
1. Implement ANH for Android following [this](https://github.com/Azure/azure-notificationhubs-android)
2. Implement ACS for Android following [this](https://docs.microsoft.com/en-us/azure/communication-services/quickstarts/voice-video-calling/calling-client-samples?pivots=platform-android#push-notifications)
3. Try receive incoming call push notification.
```
{callId=75dc76a5-b1f5-420f-ae48-774e5b9c22f5, callerId=acs:918597be-3f29-4680-a91a-b4f5e6758a17_00000009-0c4e-ed6c-b5bb-a43a0d00c613, eventType=107, convoId=acs:918597be-3f29-4680-a91a-b4f5e6758a17_00000009-0c4e-ed6c-b5bb-a43a0d00c613, servicePayload={"cp":"H4sIAAAAAAAEAL1UyY7cNhD9Fx18MlviTjYwCGzDmRiGx2ucTC5BkSxpOKOWZInddmPQ\/x6ql3HQmIOBINaBApeqV+\/Vcl94aNurPsU6ekix74rlfVGP\/Wr+x1AsC7MEPy0tNdJqh4TXzBKhTEXAUiBO1BKVlgao\/rs6fJZUXiDBoDxx0jkCgkMVqsoryounRYjT0ML2Clb4v\/jHLgx97NKrOXxplVBKGQKUGSIq64hxLFsDAHNKOKZMtmmha9bQ4N4Gu3wywJiijwMcHSmgnDlviTNMEyG9IQ4tZJeuoqHCjBOy2U0MIdsva2gn3D0tUv8flDREa00J0pAX7jWR9sAUuRLnSnbrtj0jf3RTkUeW03dG\/uDlnDxUoKyWNcHMngimagLOMWIsBAlMoNT8EfJt7O6mmT+kBP4mO7pJaZiWZek9uR0QSKUX3i+mu+2AC9+vlkLwfFduaFn34x2GkrtgqDY1YWBy\/jRVxErDiXMeg9ChZrUpmSgjzaspD0C\/xAtKczwrDBGeddNXHDO29xl4BXNux74ZcZp+QkAnqFNII96iTz8B+AB0gl2H4dMI3TT04wyetxlasgWlYsHZgqslF9qUxe4o2ou+S9ilUz3k969xO2dyylzymNjvil\/\/eP9brBt2+Um+vZlgc\/XnS2Gai4sMmMvnDmeot9fvjb6V15e5fr68RHnN1Aec3+xmMXL5epxOMJs+elxBbD9iSrFr5trZ5Xe+7zY4TvsB9arbxPQwqv59M8c89m27T\/aDvPnBUWC7mDfnEuej0n9+HfzXzV\/t79uPTHPTXbYvrpusnXiCF4prySqrOFeMWs3krGec3qzbFN\/lPtke630+fD72EDxM6XsPBHTr5kHO+\/3A3feUlsFrBZI4WsvcU1XuKRRzy+d2ctYzVstH5tAPtWLGbcZ+Pexxvx3TuPsHwghGqe4FAAA="}, displayName=8:acs:918597be-3f29-4680-a91a-b4f5e6758a17_00000009-0c4e-ed6c-b5bb-a43a0d00c613, recipientId=acs:918597be-3f29-4680-a91a-b4f5e6758a17_00000008-7771-e1d1-e3c7-593a0d00e364}
```
4. Change below code from documentation as the handlePushNotification function parameter changed from IncomingCallInformation to PushNotificationInfo(updated from 1.0.0-beta.8 -> 1.0.0-beta.9)
```
try {
    IncomingCallInformation notification = IncomingCallInformation.fromMap(pushNotificationMessageDataFromFCM);
    Future handlePushNotificationFuture = callAgent.handlePushNotification(notification).get();
}
catch(Exception e) {
    System.out.println("Something went wrong while handling the Incoming Calls Push Notifications.");
}
```
to
```
try {
    PushNotificationInfo notification = PushNotificationInfo.fromMap(pushNotificationMessageDataFromFCM);
    Future handlePushNotificationFuture = callAgent.handlePushNotification(notification).get();
}
catch(Exception e) {
    System.out.println("Something went wrong while handling the Incoming Calls Push Notifications.");
}
```
4. See error occurs when parsing notification payload with PushNotificationInfo.fromMap(payload)

**Expected behavior**
Expected to get event raising on onPropertyChanged for incoming call after handling push notification.

**Smartphone (please complete the following information):**
 - Device: samsung SM-G950N
 - OS: Android 8.0
 - Version 29

**Additional context**
There's no document about Android Kotlin for ACS please update.

**Logs**
```
2021-03-26 13:43:35.184 763-941/com.cloudhospital.int E/AndroidRuntime: FATAL EXCEPTION: Firebase-Messaging-Intent-Handle
    Process: com.cloudhospital.int, PID: 763
    java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/gson/JsonObject;
        at com.azure.android.communication.calling.PushNotificationInfo.extractNotificationMessage(PushNotificationInfo.java:136)
        at com.azure.android.communication.calling.PushNotificationInfo.fromMap(PushNotificationInfo.java:164)
        at com.contoso.service.CHFirebaseMessagingService.onMessageReceived(CHFirebaseMessagingService.kt:24)
        at com.google.firebase.messaging.FirebaseMessagingService.dispatchMessage(com.google.firebase:firebase-messaging@@21.0.1:13)
        at com.google.firebase.messaging.FirebaseMessagingService.passMessageIntentToSdk(com.google.firebase:firebase-messaging@@21.0.1:8)
        at com.google.firebase.messaging.FirebaseMessagingService.handleMessageIntent(com.google.firebase:firebase-messaging@@21.0.1:3)
        at com.google.firebase.messaging.FirebaseMessagingService.handleIntent(com.google.firebase:firebase-messaging@@21.0.1:3)
        at com.google.firebase.messaging.EnhancedIntentService.lambda$processIntent$0$EnhancedIntentService(Unknown Source:1)
        at com.google.firebase.messaging.EnhancedIntentService$$Lambda$0.run(Unknown Source:6)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:6)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.gson.JsonObject" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.cloudhospital.int--XvUJ0fvCDoQkCUiTZHwbA==/base.apk"],nativeLibraryDirectories=[/data/app/com.cloudhospital.int--XvUJ0fvCDoQkCUiTZHwbA==/lib/arm64, /data/app/com.cloudhospital.int--XvUJ0fvCDoQkCUiTZHwbA==/base.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.azure.android.communication.calling.PushNotificationInfo.extractNotificationMessage(PushNotificationInfo.java:136) 
        at com.azure.android.communication.calling.PushNotificationInfo.fromMap(PushNotificationInfo.java:164) 
        at com.contoso.service.CHFirebaseMessagingService.onMessageReceived(CHFirebaseMessagingService.kt:24) 
        at com.google.firebase.messaging.FirebaseMessagingService.dispatchMessage(com.google.firebase:firebase-messaging@@21.0.1:13) 
        at com.google.firebase.messaging.FirebaseMessagingService.passMessageIntentToSdk(com.google.firebase:firebase-messaging@@21.0.1:8) 
        at com.google.firebase.messaging.FirebaseMessagingService.handleMessageIntent(com.google.firebase:firebase-messaging@@21.0.1:3) 
        at com.google.firebase.messaging.FirebaseMessagingService.handleIntent(com.google.firebase:firebase-messaging@@21.0.1:3) 
        at com.google.firebase.messaging.EnhancedIntentService.lambda$processIntent$0$EnhancedIntentService(Unknown Source:1) 
        at com.google.firebase.messaging.EnhancedIntentService$$Lambda$0.run(Unknown Source:6) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:6) 
        at java.lang.Thread.run(Thread.java:764) 

```

**Comment by hyounoo** [comment]
Found out that I need gson dependency on my gradle.
implementation 'com.google.code.gson:gson:2.8.6'

**Comment by kagbakpem** [comment]
Thanks for reporting this @hyounoo. Have the gson dependency solved the issue for you?

**Comment by kagbakpem** [comment]
This has been solved since release of 1.0.0. Thanks
